aws/lambda-with-other-services/lambda-dynamodb/lambda-dynamodb.md

```diff
@@ -1,3 +1,298 @@
+        {
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:PutItem"
+            ],
+            "Resource":[
+		"Table’s ARN goes here"
+            ]
+        }
+    ]
+}
+```
+
+![alt text](images/4.png)
+
+Click "Review Policy," give the policy a name and description, and click "Create Policy."
+
+![alt text](images/5.png)
+
+Next we'll need to create the function's role and attach the policy we've just created. Select the "Roles" tab on the side navbar and click "Create role."
+
+![alt text](images/6.png)
+
+Select Lambda as the service that will use this role and click "Next: Permissions."
+
+![alt text](images/7.png)
+
+Search for the policy you've just created in the search box, check the box next to it, and click "Next: Review."
+
+![alt text](images/8.png)
+
+Give the role a name and description, and click the button to create the role.
+
+![alt text](images/9.png)
+
+## Creating the Lambda Function
+
+Recall that in our table, the primary key is a string that we have named "id". We will be generating these id's using a Node package. This means we will need to create the function locally and install the package to it before deploying it to Lambda.
+
+Create a directory for the function, open a terminal in the directory, and initialize the project using pnpm.
+
+We will be creating unique id's for the table using the node package [uuid](https://www.npmjs.com/package/uuid). A uuid is a [universally unique identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier), which is an encoded number generated based on unique components such as a timestamp, a random number, and network information about the host generating the uuid. Using uuid's is an effective way to ensure that we will be getting unique values, which is important for the primary key of a table. The terminal command to install the package to the project is:
+
+```
+$ pnpm install uuid
+```
+
+Create an index.js file in the project and open it in your editor. This is where we will be writing the function itself.
+
+First add the dependencies. We need to import the uuid package we've just installed. We also need to require the AWS SDK (which all Lambda functions have access to, so we don't have to install it). The SDK will then allow us to create an instance of the DynamoDB document client.
+
+```javascript
+const AWS = require('aws-sdk');
+const docClient = new AWS.DynamoDB.DocumentClient({region: 'us-west-2'});
+const uuid4 = require('uuid/v4');
+```
+
+This docClient instance is a standard way to target DynamoDB. It provides a simple way to create JavaScript sets. As we are specifying a region here, our function will be able to write to the table even if they are in different AWS regions.
+
+Now for the function itself. As with all Lambda functions, we will use an ```exports.handler``` function that has three arguments: event, context, and callback.
+
+```javascript
+exports.handler = function(event, context, callback) {
+
+	}
```

Comment:
This curly brace is unnecessary. You should also put `const docClient = new AWS.DynamoDB.DocumentClient({region: 'us-west-2'});` in here